packages/lesswrong/components/form-components/FormComponentDateTime.tsx

```diff
@@ -269,16 +296,11 @@ const DatePicker = ({label, name, value, below, onChange, onClose, classes}: {
         inputProps={{
           name:name,
           autoComplete:"off",
-          className:classes.input
-        }}
-        onChange={(newDate: Moment) => {
-          // newDate argument is a Moment object given by react-datetime.
-          onChange(newDate.toDate())
-        }}
-        onBlur={(newDate: Moment) => {
-          // newDate argument is a Moment object given by react-datetime.
-          onClose?.(newDate.toDate())
+          className: classNames(classes.input, {[classes.error]: error}),
+          ...(valueJustCleared && {value: undefined}),
```

Comment:
To explain the field-clearing bug: The DateTimePicker has it's `value`, which is the value set in the form state. Then it has an internal `<input>` which has its own value which is controlled by the DateTimePicker. When you type an invalid date it only changes the `value` of the `<input>` but not the value of the DateTimePicker/form-state.

Clicking the clear button just sets the form state value to null. If you have just cleared it, and then typed an invalid date, the form state value will already be null, so there is no way to tell that the field has just been cleared. Adding a callback for when a field is cleared would be the best way to deal with this, but that's pretty fiddly and in practice it seems like people rarely type into this input